dd-sdk-android-tv/src/main/kotlin/com/datadog/android/tv/LeanbackViewAttributesProvider.kt

```diff
@@ -0,0 +1,119 @@
+import com.datadog.android.rum.tracking.ViewAttributesProvider
+
+/**
+ * Provides extra attributes for the touch target View.
+ * Those special attributes are specifically related with Android TV [Action].
+ * @see [ACTION_TARGET_ACTION_ID]
+ * @see [ACTION_TARGET_LABEL1]
+ * @see [ACTION_TARGET_LABEL2]
+ */
+class LeanbackViewAttributesProvider :
+    ViewAttributesProvider {
+
+    // region ViewAttributesProvider
+
+    override fun extractAttributes(
+        view: View,
+        attributes: MutableMap<String, Any?>
+    ) {
+        // traverse the target parents
+        var parent = view.parent
+        var child: View? = view
+        while (parent != null) {
+            if (parent is RecyclerView && child != null && isDirectChildOfRecyclerView(child)) {
+                parent.findAction(child)?.let {
+                    attributes[ACTION_TARGET_ACTION_ID] = it.id
+                    it.label1?.let {
+                        attributes[ACTION_TARGET_LABEL1] = it
+                    }
+                    it.label2?.let {
+                        attributes[ACTION_TARGET_LABEL2] = it
+                    }
+                }
+                break
+            }
+            child = parent as? View
+            parent = parent.parent
+        }
+    }
+
+    // endregion
+
+    // region Object
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return javaClass.hashCode()
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun isDirectChildOfRecyclerView(child: View): Boolean {
+        return child.layoutParams is RecyclerView.LayoutParams
+    }
+
+    private fun RecyclerView.ViewHolder.isBridgeAdapterViewHolder(): Boolean {
+        return ItemBridgeAdapter.ViewHolder::class.java.isAssignableFrom(this::class.java)
+    }
+
+    private fun RecyclerView.findActionViewHolder(view: View): ItemBridgeAdapter.ViewHolder? {
+        val viewHolder = findContainingViewHolder(view)
+        if (viewHolder?.isBridgeAdapterViewHolder() == true) {
+            return viewHolder as ItemBridgeAdapter.ViewHolder
+        }
+        return null
+    }
+
+    private fun RecyclerView.findAction(view: View): Action? {
+        return findActionViewHolder(view)?.action()
+    }
+
+    private fun ItemBridgeAdapter.ViewHolder.action(): Action? {
+        if (Action::class.java.isAssignableFrom(item::class.java)) {
+            return item as Action
+        }
+        return null
+    }
+
+    // endregion
+
+    companion object {
+
+        /**
+         * The action id assigned for the current target.
+         * @see [Action]
+         */
+        const val ACTION_TARGET_ACTION_ID: String = "action.target.actionid"
+
+        /**
+         * The action label1 assigned for the current target.
+         * @see [Action]
```

Comment:
Why did leanback name those label1 and label2? Can we add a bit more documentation on our side  

```suggestion
         * The  label displayed on the first line of the current target Action.
         * @see [Action]
```

Reply:
They don't say anything, I think was just they way they chose to name them.